### PR TITLE
symbols: fix possible nil pointer dereference when GraphQL query

### PR DESCRIPTION
**Pull request by sashaostrikov**
contains no "query" clause in `symbols` resolver.

This is not guaranteed anywhere up the code stack (`query` is not a mandatory GraphQL parameter), that's why I added this check. I came across it during local testing with a query like:
```
query CommitSymbols($repoName: String!, $revision: String!) {
	repository(name: $repoName) {
		commit(rev: $revision) {
            symbols {
				nodes {
					name
				}
			}
		}
	}
}
```

Test plan:
Existing CI, because adding a unit test for `searchZoekt` looks non-trivial and the change is tiny.

### CODE REVIEW

**Review by ryanslade — APPROVED** [review]
LGTM, maybe add a test that causes the old code to panic?

**Comment by sashaostrikov** [comment]
> LGTM, maybe add a test that causes the old code to panic?

I actually did it, but as soon as NPE is fixed, it becomes really hard to stub everything so that `searchZoekt` doesn't fail.

Do you think it is worth adding a test which just checks that an error is not the nil panic?
@ryanslade

**Comment by ryanslade** [comment]
> > LGTM, maybe add a test that causes the old code to panic?
> 
> I actually did it, but as soon as NPE is fixed, it becomes really hard to stub everything so that `searchZoekt` doesn't fail.
> 
> Do you think it is worth adding a test which just checks that an error is not the nil panic? @ryanslade

What I've done in some tests is that once I'm passed the code I want to check I return a "short circuit" error and check for that:

https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@d21ad9ab836a7b3458bcc0ed59cc7128bf380f82/-/blob/cmd/frontend/graphqlbackend/user_emails_test.go?L147:29

**Comment by sashaostrikov** [comment]
> > > LGTM, maybe add a test that causes the old code to panic?
> > 
> > 
> > I actually did it, but as soon as NPE is fixed, it becomes really hard to stub everything so that `searchZoekt` doesn't fail.
> > Do you think it is worth adding a test which just checks that an error is not the nil panic? @ryanslade
> 
> What I've done in some tests is that once I'm passed the code I want to check I return a "short circuit" error and check for that:
> 
> https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@d21ad9ab836a7b3458bcc0ed59cc7128bf380f82/-/blob/cmd/frontend/graphqlbackend/user_emails_test.go?L147:29

I think I'll try to do the same thing, I just need to find a thing that can be mocked there 👍🏻

File: internal/search/symbol/symbol.go [hunk 1]
```diff
@@ -71,8 +71,8 @@ func filterZoektResults(ctx context.Context, checker authz.SubRepoPermissionChec
 }
 
 func searchZoekt(ctx context.Context, repoName types.MinimalRepo, commitID api.CommitID, inputRev *string, branch string, queryString *string, first *int32, includePatterns *[]string) (res []*result.SymbolMatch, err error) {
-	raw := *queryString
-	if raw == "" {
+	var raw string
+	if queryString == nil || *queryString == "" {
```
Comment (mrnugget) [inline internal/search/symbol/symbol.go:75]: So previously we used the `*queryString` value, now we don't anymore. I think what you want is

```go
var raw string
if queryString != nil {
   raw = *queryString
}
if raw == "" {
   raw = ".*"
}
```

**Review by mrnugget — CHANGES_REQUESTED** [review]

File: internal/search/symbol/symbol.go (hunk 1 above)
Comment (sashaostrikov) [reply internal/search/symbol/symbol.go:75]: whoa, that would've been really bad to get merged...
Thanks for catching that!

**Review by mrnugget — APPROVED** [review]
I feel like the test/mock setup is a bit overkill for a nil check, but fine with me.

**Comment by sashaostrikov** [comment]
> I feel like the test/mock setup is a bit overkill for a nil check, but fine with me.

Probably one day I will come up with a unit test for this function (which would have caught my bug, btw), so the mock is more used :)

**Pull request merged**